packages/react/src/components/TextEllipsis/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Ref, useEffect, useState } from 'react';
+import classnames from 'classnames';
+import useSharedRef from '../../utils/useSharedRef';
+import Tooltip, { type TooltipProps } from '../Tooltip';
+
+interface TextEllipsisProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: string;
+  maxLines?: number;
+  as?: React.ElementType;
+  refProp?: string;
+  tooltipProps?: Omit<TooltipProps, 'target' | 'association'>;
+}
+
+const TextEllipsis = React.forwardRef(
+  (
+    {
+      className,
+      children,
+      maxLines,
+      as,
+      tooltipProps,
+      ...props
+    }: TextEllipsisProps,
+    ref: Ref<HTMLElement>
+  ) => {
+    let Element: React.ElementType<any> = 'div';
+    const sharedRef = useSharedRef<HTMLElement>(ref);
+    const [showTooltip, setShowTooltip] = useState(false);
+
+    if (as) {
+      Element = as;
+    } else if (showTooltip) {
```

Comment:
So is the expectation that if you pass `as`, it _has_ to be interactive (either semantically or the caller is responsible themselves for setting the proper interactive attributes)? Might be good to call that out in the docs, since someone might pass `p` or some other non-interactive as the `as`.

Reply:
It's already called out a bit:

> If your component is a Button, Link, or some other kind of interactive element with a `tabIndex`, you _must_ provide your component as a Polymorphic component using the `as` prop to avoid nesting interactive elements

IMO authors _should not_ be passing in a non-interactive element. We can make that indication stronger if you think it's needed, but we have no way of validating that the component that they pass in is interactive.

Reply:
As a nit, I would suggest documenting that the expectation is that `as` should be interactive, even if we can't verify it.

Reply:
Added a clarifying note.